utils/cloud_cleanup/aws/clean_aws.py

```diff
@@ -0,0 +1,346 @@
+        ka_lt = keep_alive_instance_launch_time(instance)  # keep:X_HOURS
+        ka_min = keep_alive_instance_duration(instance, duration)
+
+        if kalive or ka_lt or ka_min:
+            if kalive or ka_lt:
+                keep_alive = keep_alive + 1
+                if VERBOSE:
+                    print_instance(instance, "keep alive")
+            if ka_lt:
+                expiring = expiring + 1
+            running = running + 1
+            debug("will not be stopped")
+        else:
+            expired.append(instance)
+
+    return running, keep_alive, expiring, expired
+
+
+def clean_instances(region_name, duration):
+    print("cleaning region %s instances" % region_name)
+    running, keep_alive, expiring, expired = scan_region_instances(region_name, duration)
+    terminated = 0
+    # stop instances
+    for instance in expired:
+        if is_running(instance):
+            if should_terminate(instance):
+                print_instance(instance, "terminating...")
+                terminated = terminated + 1
+                remove_protection(instance)
+                terminate_instance(instance)
+            else:
+                print_instance(instance, "stopping...")
+                remove_protection(instance)
+                stop_instance(instance)
+
+    # blind wait stop timeout
+    if len(expired) > 0 and not DRY_RUN:
+        print("region %s: waiting %d instances to stop" % (region_name, len(expired)))
+        time.sleep(WAIT)
+
+    # check instances are stopped. if they are not - terminate them
+    for instance in expired:
+        instance.reload()
+        if not is_stopped(instance) and not is_terminated(instance):
+            print_instance(instance, "terminating...")
+            terminate_instance(instance)
+
+    print("region %s: stopped %d, terminated %d, kept alive %d and expiring %d, running %d instances" % (
+        region_name, len(expired) - terminated, terminated, keep_alive, expiring, running))
+
+
+def keep_alive_volume(volume):
+    # checking tags
+    if volume.tags is None:
+        return False
+    for tag in volume.tags:
+        if tag['Key'] == 'keep' and tag['Value'] == 'alive':
+            return True
+    return False
+
+
+def delete_volume(volume):
+    try:
+        print_volume(volume, "deleting")
+        if not DRY_RUN:
+            volume.delete()
+    except Exception:  # pylint: disable=broad-except
+        pass
+
+
+def print_volume(volume, msg):
+    print("volume %s %s" % (volume.id, msg))
+
+
+def clean_volumes(region_name):
+    print("cleaning region %s volumes" % region_name)
+    ec2 = boto3.resource('ec2', region_name=region_name)
+
+    volumes = ec2.volumes.all()
+
+    count_kept_volume = 0
+    count_deleted_volume = 0
+    for volume in volumes:
+        debug("checking volume %s %s %s %s" % (volume.id, volume.volume_id, volume.state, volume.tags))
+        keep_alive = keep_alive_volume(volume)
+        if keep_alive or volume.state == "in-use":
```

Comment:
https://github.com/scylladb/cloud-devops/issues/27

still need to be fixed, exactly here...

Reply:
fixed (I suppose) - added to `keep_alve_volume` def.

Reply:
o.k. didn't saw it

Reply:
you couldn't as it wasn't that time in place :)